### PR TITLE
Drop aggregate-error and require node >=16

### LINKED ISSUE #59: Browser Compatibility with `^5.0.0`
After upgrading to ^5.0.0, vite got an error: 
```
browser-external:os:3 Uncaught Error: Module "os" has been externalized for browser compatibility and cannot be accessed in client code.
    at Object.get (browser-external:os:3:11)
    at index.js:6:27
```

After some digging, this is related to `aggregate-error  ^4.0.0`'s dep `clean-stack ^4.0.0` 

[`import os from 'os'` (index.js#L1)](https://github.com/sindresorhus/clean-stack/blob/b09f9796fc87667c8b58631b91b5d41ce5dccc12/index.js#L1)

I wonder could `homedir` be implemented in a browser-friendly way?

I don't  know which repo is the most suitable place to open this issue, please transfer to the right place if needed.

Thank you!

### LINKED ISSUE #64: Remove `aggregate-error` dependency
Node 14 will reach EOL in 6 weeks, and Node 16 has native [`AggregateError`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AggregateError) so the dependency should be dropped [as outlined here](https://github.com/sindresorhus/clean-stack/issues/28#issuecomment-917834223) for the release that will bump min Node version to 16.

Related: https://github.com/sindresorhus/clean-stack/issues/28
Related: https://github.com/sindresorhus/p-map/issues/59

### PR DESCRIPTION
**Pull request by furudean**
See rationale in #64

fixes #64
fixes #59

### CODE REVIEW

**Comment by furudean** [comment]
The linter wants an error message string. I guess this only gets picked up on the native AggregateError. Should we just ignore this, or add a message? @sindresorhus

**Comment by sindresorhus** [comment]
Ignore it

**Comment by silverwind** [comment]
Could wait until 2023-04-18 with this when Node 20 releases, so support for 20 and requirement of 16 can be done in one go.

**Comment by sindresorhus** [comment]
Yes, I don't plan to merge this until then.

File: index.js [hunk 1]
```diff
@@ -99,7 +97,9 @@ export default async function pMap(
 
 				if (resolvingCount === 0 && !isResolved) {
 					if (!stopOnError && errors.length > 0) {
+						/* eslint-disable unicorn/error-message */
 						reject(new AggregateError(errors));
+						/* eslint-enable unicorn/error-message */
```
Comment (silverwind) [inline index.js:102]:
```suggestion
						reject(new AggregateError(errors)); // eslint-disable-line unicorn/error-message
```

**Comment by silverwind** [comment]
Time for merge?

**Pull request merged**